include/net/net_pkt.h

```diff
@@ -75,13 +75,12 @@ struct net_pkt {
 
 #if defined(CONFIG_NET_TCP)
 	sys_snode_t sent_list;
-#endif
-
 	u8_t sent_or_eof: 1;	/* For outgoing packet: is this sent or not
 				 * For incoming packet of a socket: last
 				 * packet before EOF
 				 * Used only if defined(CONFIG_NET_TCP)
 				 */
+#endif
```

Comment:
If you do this, then the _unused bit count would be wrong if TCP is not enabled. Better to always enable the sent_or_eof bit field, it is also mentioned in the comment that it is only **used** if TCP is enabled.